sklearn/utils/_plotting.py

```diff
@@ -208,13 +218,30 @@ def _validate_curve_kwargs(
         # Ensure `curve_kwargs` is of correct length
         if isinstance(curve_kwargs, Mapping):
             curve_kwargs = [curve_kwargs] * n_curves
+        elif curve_kwargs is None:
+            curve_kwargs = [{}] * n_curves
+
+        if default_curve_kwargs is None:
+            default_curve_kwargs = {}
+        if default_multi_curve_kwargs is None:
+            default_multi_curve_kwargs = {}
+        # This should never happen, as we would not pass the same keys to both
+        # default kwargs
+        common_keys = set(default_curve_kwargs).intersection(default_multi_curve_kwargs)
+        if len(common_keys) > 0:
+            raise ValueError(
+                "`default_curve_kwargs` and `default_multi_curve_kwargs` both contain "
+                f"the keys {common_keys}. The same kwargs cannot be passed to both "
+                "defaults."
+            )
 
-        default_multi_curve_kwargs = {"alpha": 0.5, "linestyle": "--", "color": "blue"}
-        if curve_kwargs is None:
-            if n_curves > 1:
-                curve_kwargs = [default_multi_curve_kwargs] * n_curves
-            else:
-                curve_kwargs = [{}]
+        if n_curves > 1:
+            default_curve_kwargs_ = {
+                **default_multi_curve_kwargs,
+                **default_curve_kwargs,
+            }
+        else:
+            default_curve_kwargs_ = default_curve_kwargs
```

Comment:
Following my previous comment, this would just be
```suggestion
        default_curve_kwargs_ = default_curve_kwargs
        if n_curves > 1:
            default_curve_kwargs_.update(default_multi_curve_kwargs)
```